File: backend/src/services/captionGenerator.js
export function buildCaption({ brand, headline, subheadline, ctaText, hashtags }) {
  const lines = [];

  lines.push(`${brand.logoEmoji} ${headline || brand.tagline}`);
  if (subheadline) {
    lines.push('');
    lines.push(subheadline);
  }

  lines.push('');
  lines.push(`📲 ${ctaText || brand.defaultCta}`);
  lines.push('');
  lines.push(buildHashtags(brand, hashtags));

  return lines.join('\n');
}

export function buildHashtags(brand, extra = []) {
  const tags = new Set([...brand.hashtags, ...extra]);
  return [...tags].map((t) => `#${t.replace(/\s+/g, '')}`).join(' ');
}

File: android/app/src/main/java/com/thecompass/contentstudio/CaptionGenerator.kt
package com.thecompass.contentstudio

object CaptionGenerator {
    fun buildCaption(brand: Brand, headline: String, subheadline: String?, ctaText: String?, hashtags: List<String>): String {
        val lines = mutableListOf<String>()
        lines += "${brand.logoEmoji} ${headline.ifBlank { brand.tagline }}"
        if (!subheadline.isNullOrBlank()) {
            lines += ""
            lines += subheadline
        }
        lines += ""
        lines += "📲 ${ctaText?.takeIf { it.isNotBlank() } ?: brand.defaultCta}"
        lines += ""
        lines += buildHashtags(brand, hashtags)
        return lines.joinToString("\n")
    }

    fun buildHashtags(brand: Brand, extra: List<String> = emptyList()): String {
        val tags = (brand.hashtags + extra).distinct()
        return tags.joinToString(" ") { "#${it.replace(Regex("\\s+"), "")}" }
    }
}
